app/ece.js

```diff
@@ -0,0 +1,237 @@
+require('buffer');
+
+const NONCE_LENGTH = 12;
+const TAG_LENGTH = 16;
+const KEY_LENGTH = 16;
+const MODE_ENCRYPT = 'encrypt';
+const MODE_DECRYPT = 'decrypt';
+
+const encoder = new TextEncoder();
+
+function generateSalt(len) {
+  const randSalt = new Uint8Array(len);
+  window.crypto.getRandomValues(randSalt);
+  return randSalt.buffer;
+}
+
+/*
+mode: string, either 'encrypt' or 'decrypt'
+ikm:  Uint8Array containing key of KEY_LENGTH length
+rs:   int containing record size, optional
+salt: ArrayBuffer containing salt of KEY_LENGTH length, optional
+The transform stream takes data as UInt8Arrays on the writable side, and outputs
+UInt8Arrays on the readable side.
+*/
+export default class ECETransformer {
+  constructor(mode, ikm, rs, salt) {
+    this.mode = mode;
+    this.prevChunk;
+    this.params = {};
```

Comment:
Having a params struct rather than direct fields for key and nonce base is a little strange given the rest of this code.